Fix Frontend Failing Test: torch - math.paddle.stanh

ZJay07 opened the pull request:
<!--
This template will help you to have a meaningful PR, please follow it and do not leave it blank.
-->

# PR Description
Failing because of exp returning NaN or inf due to large x, scale_a, scale_b values. Fixed by limit those values from being too large

## Related Issue

<!--
Please use this format to link other issues with their numbers: Close #123
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

Close https://github.com/unifyai/ivy/issues/28492
Close https://github.com/unifyai/ivy/issues/28547
Close https://github.com/unifyai/ivy/issues/28546
Close https://github.com/unifyai/ivy/issues/28545
Close https://github.com/unifyai/ivy/issues/28544

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

<!--
Please mark your PR as a draft if you realise after the fact that your tests are not passing or
that your pre-commit check has some failures.

Here are some relevant resources regarding tests and pre-commit:

https://unify.ai/docs/ivy/overview/deep_dive/ivy_tests.html
https://unify.ai/docs/ivy/overview/deep_dive/formatting.html#pre-commit

-->

### Socials

<!--
If you have Twitter, please provide it here otherwise just ignore this.
-->

Ishticode reviewed (approved):
Thanks a lot for the PR @ZJay07  :)

Ishticode reviewed (changes requested):
Although, the `test_paddle_stanh` seems to still fail with paddle and torch backends. You can see that in the logs at https://github.com/unifyai/ivy/actions/runs/8232013594/job/22509022640?pr=28493
Are you able to fix for these two?

ZJay07 commented:
Yup, happy to fix it, seems like some results are producing NaN or 0 instead, do you think increasing the min and max `x` values would be sufficient?

Ishticode reviewed (changes requested):
You shold be able to reproduce the errors using the falsifying examples in the failure logs. Like one example for the failure of test_paddle_stanh is given as 
```
[array([[[-1.0000000e+00, -1.0000000e+00],
E                 [-1.0000000e+00, -1.0000000e+00],
E                 [-1.0000000e+00, -1.0000000e+00],
E                 [-1.0000000e+00, -1.0000000e+00]],
E         
E                [[-9.0909090e+00, -1.2930438e-38],
E                 [-1.8302815e-38, -2.3399685e-38],
E                 [-1.2930438e-38, -1.9542594e-38],
E                 [-1.2930438e-38, -1.0000000e+00]]], dtype=float32)]),
E       scale_a=10.0,
E       scale_b=1.0,
```
so you can write something like
```python
import ivy
import ivy.functional.frontends.paddle as ivy_paddle
ivy.set_backend("torch")
x = ivy.arrayarray([[[-1.0000000e+00, -1.0000000e+00],
[-1.0000000e+00, -1.0000000e+00],
[-1.0000000e+00, -1.0000000e+00],
[-1.0000000e+00, -1.0000000e+00]],

[-9.0909090e+00, -1.2930438e-38],
[-1.8302815e-38, -2.3399685e-38],
[-1.2930438e-38, -1.9542594e-38],
[-1.2930438e-38, -1.0000000e+00]]], dtype="float32")])
scale_a=10.0
scale_b=1.0
```
and add the same with native paddle
```python
import paddle
x = paddle.to_tensor([[[-1.0000000e+00, -1.0000000e+00],
                [-1.0000000e+00, -1.0000000e+00],
                [-1.0000000e+00, -1.0000000e+00],
                [-1.0000000e+00, -1.0000000e+00]],
               [[-9.0909090e+00, -1.2930438e-38],
               [-1.8302815e-38, -2.3399685e-38],
               [-1.2930438e-38, -1.9542594e-38],
               [-1.2930438e-38, -1.0000000e+00]]], dtype="float32")
scale_a=10.0
scale_b=1.0
print(paddle.stanh(x, scale_a, scale_b))
```
and you will see the value difference and why the test failed as ivy's paddle frontend's behaviour differs from native paddle.

My point is try to reconstruct the failing example from the falsifying examples provided in the logs and this way you can always reproduce and debug quickly. Thank you very much :)

ZJay07 commented:
Hey @Ishticode, I just updated the backend paddle code to accept `stanh` and re-cast it back to the same dtype as x as per paddle docs, also added some factors to `scale_a` and `scale_b` that is similar to `addmm` in the same file

Ishticode reviewed (approved):
now it seems to behave well :) haha
Thanks @ZJay07 for the fix
